content/post/review-programs-with-graphs.md

```diff
@@ -0,0 +1,322 @@
+Learning model used in the paper — **“Gated Graph Neural Networks”**. 🕵
+
+### History: chemistry -> message passing -> ML on Code
+
+In the official [model's repository README](https://github.com/Microsoft/gated-graph-neural-network-samples#gated-graph-neural-networks)
+ the authors share that the inspiration for this work comes from another field of ML research: Quantum Chemistry.
+
+Interestingly enough, in a few recent talks, Jeff Dean while talking about most exciting advances in ML applications
+mentions Quantum Chemistry as well:
+
+* [TWiML & AI](https://medium.com/@twimlai) podcast episode [“Systems and Software for Machine Learning at Scale”](https://twimlai.com/twiml-talk-124-systems-software-machine-learning-scale-jeff-dean/)
+
+* [Matroid](https://www.matroid.com/) conference [Scaled Machine Learning](http://scaledml.org/) talk [“Systems and Machine Learning”](https://www.matroid.com/scaledml/2018/jeff.pdf)
+
+{{% caption src="https://cdn-images-1.medium.com/max/3804/1*RHbRW58oHZOp4vFqwDTNKg.png" title="ML application in quantum physics" %}}
+ML application in quantum physics. Source: Jeff Dean slides
+{{% /caption %}}
+
+The fundamental idea is well-understood — given a [Schrödinger equation](https://en.wikipedia.org/wiki/Schr%C3%B6dinger_equation),
+one can get information about the state of a single particle, thus solving it for composition would allow to
+model properties of more complex structures, including molecules and in general, a solid state matter. But
+solving many-body Schrödinger equation requires huge computational efforts.
+
+Instead, a [Nobel prize winning](https://www.nobelprize.org/nobel_prizes/chemistry/laureates/1998/) modeling
+approach called [Density Functional Theory](https://en.wikipedia.org/wiki/Density_functional_theory) can be
+applied, reducing the problem of many-body interacting system to a series of single-body problems and although still
+slow, it is highly valuable for many tasks in physics, chemistry and material science. Many approximation methods
+have been developed to make this more feasible by getting estimates instead of exact answers.
+
+### Neural Networks for predicting properties of molecules
+
+So in 2017, several researchers at Google Brain Residence program spent time applying recent ML optimization methods
+for predicting properties of molecules:
+
+* many [software simulators](https://en.wikipedia.org/wiki/List_of_quantum_chemistry_and_solid-state_physics_software)
+already exist for DFT, but are slow and still computationally intensive
+
+* in 2014 those simulators were used to build a [reference dataset — QM9](http://quantum-machine.org/datasets/),
+  suitable for supervised learning
+
+* a new “featurization” method was proposed, for looking at molecular structure as a graph: atoms as nodes and bonds as edges
+
+* a new variation of the Gated Graph Neural Network architecture was proposed, particularly suited for summarizing
```

Comment:
a new variation? you haven't explained what those are yet 😕

Reply:
the thing is - this paper itself does not explain basic Gated Graph Neural Network architecture.
But I have linked to the original paper now https://arxiv.org/abs/1511.05493